keyboards/bm40hsrgb/keymaps/gabustoledo/keymap.c

```diff
@@ -0,0 +1,161 @@
+/* Copyright 2021 Gabriel Bustamante Toledo
+  * 
+  * This program is free software: you can redistribute it and/or modify 
+  * it under the terms of the GNU General Public License as published by 
+  * the Free Software Foundation, either version 2 of the License, or 
+  * (at your option) any later version. 
+  * 
+  * This program is distributed in the hope that it will be useful, 
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+#include QMK_KEYBOARD_H
+
+enum layers { BASE1, BASE2, MEDIA, NAV, MOUSE, SYM, NUM, FUN };
+
+enum {
+    TD_MEDIA, TD_SCREEN,
+};
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_MEDIA] = ACTION_TAP_DANCE_DOUBLE( KC_MPLY , KC_MNXT ),
+    [TD_SCREEN] = ACTION_TAP_DANCE_DOUBLE( (G(S(KC_S))) , S(C(KC_4)) ),
+};
+
+void rgb_matrix_indicators_user(void) {
+
+	if(IS_LAYER_ON(BASE2)) {
```

Comment:
You might want to use a switch block for these instead, it's much cleaner.

Reply:
I don't know how to clean the code further. Also, I don't know how to detect the layer on the switch. If you have an example, show me.

Reply:
```c
switch (get_highest_layer(layer_state)) {
    case BASE2:
        // ...
        break;
    case MEDIA:
        // ...
        break;
}
```
etc.